src/Catel.Core/Catel.Core.Shared/Data/ModelEqualityComparer.cs

```diff
@@ -159,13 +161,7 @@ public override bool Equals(ModelBase x, ModelBase y)
         /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
         public override int GetHashCode(ModelBase obj)
         {
-            if (obj == null)
-            {
-                return 0;
-            }
-
-            var objType = obj.GetType();
-            return objType.FullName.GetHashCode();
+            return RuntimeHelpers.GetHashCode(obj);
```

Comment:
This is a big question what this will change ... but hashing algorithmes need different hashes to work as intended.